### PR TITLE
start talking about containers

### PR DESCRIPTION
**Pull request by bengerman13**
## Changes proposed in this pull request:
- start talking about containers

## security considerations
This is all about security.

### CODE REVIEW

**Comment by davemcorwin** [comment]
This assumes that the scanning that can be done by Docker Hub or ECR would not be sufficient?

**Comment by bengerman13** [comment]
> This assumes that the scanning that can be done by Docker Hub or ECR would not be sufficient?

Docker Hub is not FedRAMP approved, so we definitely need validation independent of them.

I don't know of any scanning done by ECR - is there something you're looking at there?

**Comment by davemcorwin** [comment]
I haven't looked into it too deeply, but there is [basic scanning](https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-scanning-basic.html) (free) and [enhanced scanning](https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-scanning-enhanced.html) (paid) which uses [amazon inspector](https://docs.aws.amazon.com/inspector/latest/user/enable-disable-scanning-ecr.html)

**Comment by bengerman13** [comment]
I was assuming we'd stand up a dedicated testing tool, since that'd been what we'd talked about previously, but I think we should check if either of these scans are usable here (both technically suitable and compliance-satisfying).

File: adr/ADR-0007-container-scanning-and-validation.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+We'd add one or more concourse pipelines that 
+- pull images
+- scan them using our to-be-identified image scanner
+- if the image passes as usable, adds the image hash to a trusted list in a global store
+- if the image has any findings, puts reports/summaries of those findings somewhere to be consumed by the team
+
+Concourse pipelines that use images would need to reference images by _hash_ rather than _tag_
+
+### Adding a new image looks like
```
Comment (mheadd) [inline adr/ADR-0007-container-scanning-and-validation.md:35]:
```suggestion
_Adding a new image looks like_
```

Comment (mheadd) [reply adr/ADR-0007-container-scanning-and-validation.md:35]: Make format consistent with other sections.

File: adr/ADR-0007-container-scanning-and-validation.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+- add the image to the checker pipeline
+- consume the SHAs from the checker 
+
+### Updating an image version looks like
```
Comment (mheadd) [inline adr/ADR-0007-container-scanning-and-validation.md:40]:
```suggestion
_Updating an image version looks like_
```

File: adr/ADR-0007-container-scanning-and-validation.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+- get a SHA from the image checker
+
+### Revoking a scary version looks like
```
Comment (mheadd) [inline adr/ADR-0007-container-scanning-and-validation.md:44]:
```suggestion
_Revoking a scary version looks like_
```

**Review by mheadd — APPROVED** [review]
LGTM! Thanks, @bengerman13

**Pull request merged**